server/game/costs.js

```diff
@@ -358,20 +358,26 @@ const Costs = {
         };
     },
     /**
-     * Cost that requires discarding a card from hand.
+     * Cost that requires discarding a card from hand matching the passed
+     * condition predicate function.
      */
-    discardFromHand: function() {
+    discardFromHand: function(condition) {
```

Comment:
It would be preferable to default the condition here than update the existing usages. i.e.

```javascript
discardFromHand: function(condition = () => true) {
```